projects/plugins/social/src/js/components/info-section/index.js

```diff
@@ -1,33 +1,42 @@
-import { Container, Text } from '@automattic/jetpack-components';
+import { Container, Text, useBreakpointMatch } from '@automattic/jetpack-components';
```

Comment:
As our use of `useBreakpointMatch` adds a class name to the components (rather than changing the HTML structure itself), does it make sense to do it all in the CSS, or is there an advantage to using the hook?

Reply:
I think using the hook is the recommended way now for Jetpack, rather than using `@media` queries. @danielpost can you confirm this?

Reply:
Yeah, last time I used `@media` queries I was told it's better to use `useBreakpointMatch` instead. cc'ing @Automattic/jetpack-agora as the source of truth on this 😄

Reply:
I think that's correct. We prefer to use the hook to handle the different viewport sizes rather than media queries.
Also, it allows changing the component behavior depending on the viewport size.

Reply:
Great. It seems a little odd to not use CSS, but I can see that there are pros and cons.

In which case should we remove the `@media` directives that [are already in there](https://github.com/Automattic/jetpack/blob/cc91ed7ef5705d5a60b8ce99e8c221ed1279dc1f/projects/plugins/social/src/js/components/info-section/styles.module.scss#L4-L24)?

Reply:
If I'm not wrong, we should replace the queries with component handling when using the Jetpack Components (RNA).

@media queries, in some way, are decoupled with the layout handling we did: ThemeProvider, Layout, useBreakpointMatch().

For instance, it's not possible to define the media query based on CSS vars:

<img width="460" alt="Screen Shot 2022-11-07 at 12 11 15" src="https://user-images.githubusercontent.com/77539/200384590-5a35ae56-baff-4f6d-8e30-dc9651555b1d.png">

[There is a proposal](https://drafts.csswg.org/css-env-1/) that would address this situation.

Also, we could use @media queries to deal with all UI issues, less at the components level. Some components change considerably their visualization depending on the viewport size. In those cases, media-queries is not enough.

cc @renatoagds

Reply:
Thanks all! I changed the existing queries in this file too!

Reply:
> Some components change considerably their visualization depending on the viewport size. In those cases, media-queries is not enough.

Yeah that makes sense, that in some situations we want to render different components/HTML for a particular size. It seems a shame to rely on the useBreakpoint hook when it could be done in CSS (reimplementing a browser feature), but it's probably similar to the CSS-in-JS debate. If it can't be handled with media queries, grid layout etc., maybe that points to the design being too complex at different sizes? I must stop debating these things on PRs!